src/Renderer/LayeredMaterial.js

```diff
@@ -76,7 +90,7 @@ var moveElementsArraySafe = function moveElementsArraySafe(array,index, howMany,
 };
 /* eslint-enable */
 
-const LayeredMaterial = function LayeredMaterial(id) {
+const LayeredMaterial = function LayeredMaterial(material, coord) {
```

Comment:
The rest look good so I'll focus only on this part.

Here's what I had in mind:

- Replace LayeredMaterial `offsetScale` with:
```js
// express what part of the texture (source) is drawn,
// and where (destination) it is drawn.
// (mapping.source is our current `offsetScale`)
// (mapping.destination is currently baked into UV attributes of a tile)
mapping = {
    source: {
        origin: new THREE.Vector2(0, 0),
        size: 1,
    },
    destination: {
        origin: new THREE.Vector2(0, 0),
        size: 1,
    },
}
```

- replace `pitch` result from providers with a `mapping` object. Currently
  only PM textures will have a non-default destination

- replace `coord` in `function LayeredMaterial(material, coord)` with tile's mapping attribute,
  as well as it's parent's mapping attribute.
  With this change, `offsetToParent` can be written without any dependencies on WMTS logic.

What do you think ?